tasks/tasks/causemos_processors_test.py

```diff
@@ -0,0 +1,415 @@
+import json
+
+import pprint
+from pathlib import Path
+from os.path import join as path_join
+import pytest
+import numpy
+import pandas
+
+from functools import partial, reduce
+
+import warnings
+from typing import Dict, List, Tuple
+
+from tasks.causemos_processors import (
+    build_synthetic_dataset,
+    normalize_synthetic_dataset,
+    perform_pca,
+    pca_to_weights_v3,
+    pca_to_weights_v2,
+    pca_to_weights_v1,
+    iteration_func,
+    # build_tree,
+    download_datasets
+)
+
+warnings.filterwarnings('ignore')
+pp = pprint.PrettyPrinter(indent=2)
+
+root_dir = Path(__file__).resolve().parent.parent
+test_data_dir = path_join(root_dir, "test-data")
+datasets_cache_dir = path_join(test_data_dir, "datasets")
+
+sample_ND_JSON_PATH = path_join(test_data_dir, "ND-GAIN.json")
+
+with open(sample_ND_JSON_PATH) as file:
+    index_model_object = json.load(file)
+
+
+# Some of these tests are integration+side-effecty-slow
+# NOTE Run pytest with if you wish to run only unit tests or CI?:
+# $ pytest -m "not integration"
+
+# Or, to run all, from tasks proj dir:
+# python -m pytest -vvs tasks/causemos_processors_test.py
+
+
+@pytest.mark.integration
+def test_iteration_func():
+
+    retrieval_dictionary = iteration_func(index_model_object)
+
+    assert len(retrieval_dictionary) == 15
+
+    # Get the first key-value pair
+    first_key = next(iter(retrieval_dictionary))
+    first_value = retrieval_dictionary[first_key]
+
+    first_dataset_stored = first_value[0]
+
+    first_dataset_feature = first_dataset_stored["source_column"]
+    first_dataset_feature_name = first_dataset_stored["name_in_model"]
+
+    assert first_dataset_feature == "governance"
+    assert first_dataset_feature_name == "Governance indicator"
+
+@pytest.mark.integration
+@pytest.mark.side_effects
+def test_download_datasets():
+    out = download_datasets(iteration_func(index_model_object))
```

Comment:
Actually don't run these tests unless you want to really download the datasets from dojo. I think I copied the files over since I already had them! Forgot about this. I may remove this "test" entirely before merging this work (testing that files download is odd!) but it helped me during development.